### PR TITLE
Copy old process entries into aggregated file and add test for this

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**
Copy process entries for sliced files into old_processed group in aggregation file
Check these groups are created in nx_aggregation test

### CODE REVIEW

File: ParProcCo/nxdata_aggregator.py [hunk 1]
```diff
@@ -251,13 +251,22 @@ def _write_aggregation_file(self, aggregation_output: Path) -> Path:
 
             f.attrs["default"] = self.nxentry_name
 
-            for i, filepath in enumerate(self.data_files):
-                f[f"entry{i}"] = h5py.ExternalLink(str(filepath), self.nxentry_name)
+            i = 0
+            process_key = "/".join([self.nxentry_name, "process"])
```
Comment (PeterC-DLS) [inline ParProcCo/nxdata_aggregator.py:255]: Needs to check for a NXprocess group(s) (not fixed name)

File: ParProcCo/nxdata_aggregator.py [hunk 2]
```diff
@@ -251,13 +251,22 @@ def _write_aggregation_file(self, aggregation_output: Path) -> Path:
 
             f.attrs["default"] = self.nxentry_name
 
-            for i, filepath in enumerate(self.data_files):
-                f[f"entry{i}"] = h5py.ExternalLink(str(filepath), self.nxentry_name)
+            i = 0
```
Comment (PeterC-DLS) [inline ParProcCo/nxdata_aggregator.py:254]: Better to enumerate (so files with missing process groups are obvious).

Comment (VictoriaBeilsten-Edmands) [reply ParProcCo/nxdata_aggregator.py:254]: And always create old_processed group, even if empty because of all process groups being missing?

Comment (PeterC-DLS) [reply ParProcCo/nxdata_aggregator.py:254]: Probably not in that case.

File: ParProcCo/nxdata_aggregator.py [hunk 3]
```diff
@@ -130,16 +130,22 @@ def _get_default_nxgroup(self, f: Union[h5py.File, h5py.Group], class_name: str)
...
             return group_name
 
-        for group_name in f.keys():
+        group_name = self._get_group_name(f, class_name)
+        if group_name:
+            return group_name
+
+        raise ValueError(f"no {class_name} group found")
+
+    def _get_group_name(self, group: Union[h5py.File, h5py.Group], class_name: str) -> Union[str, None]:
```
Comment (PeterC-DLS) [inline ParProcCo/nxdata_aggregator.py:139]: Make this a generator so can retrieve more than just the first group

File: ParProcCo/nxdata_aggregator.py [hunk 4]
```diff
@@ -131,19 +131,19 @@ def _get_default_nxgroup(self, f: Union[h5py.File, h5py.Group], class_name: str)
             return group_name
 
         group_name = self._get_group_name(f, class_name)
-        if group_name:
-            return group_name
+        for name in group_name:
```
Comment (PeterC-DLS) [inline ParProcCo/nxdata_aggregator.py:134]: For one-off use, use `next(group_name)`

File: ParProcCo/nxdata_aggregator.py [hunk 5]
```diff
@@ -131,19 +131,19 @@ def _get_default_nxgroup(self, f: Union[h5py.File, h5py.Group], class_name: str)
...
         for group_name in group.keys():
             try:
                 class_type = group[group_name].attrs.get("NX_class", '')
                 class_type = decode_to_string(class_type)
                 if class_type == class_name:
                     group_name = decode_to_string(group_name)
-                    return group_name
+                    yield group_name
             except KeyError:
                 logging.warning(f"KeyError: {group_name} could not be accessed in {group}")
```
Comment (PeterC-DLS) [inline ParProcCo/nxdata_aggregator.py:149]: Raise stop error

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**